tests/components/amberelectric/test_services.py

```diff
@@ -0,0 +1,202 @@
+    assert first["spot_per_kwh"] == 0.01
+    assert first["start_time"] == "2021-09-21T08:30:00+10:00"
+    assert first["end_time"] == "2021-09-21T09:00:00+10:00"
+    assert first["renewables"] == 50
+    assert first["spike_status"] == "none"
+    assert first["descriptor"] == "very_low"
+
+
+@pytest.mark.usefixtures("mock_amber_client_forecasts")
+async def test_incorrect_channel_type(
+    hass: HomeAssistant,
+    general_channel_config_entry: MockConfigEntry,
+) -> None:
+    """Test error when the channel type is incorrect."""
+    await setup_integration(hass, general_channel_config_entry)
+
+    with pytest.raises(
+        vol.error.MultipleInvalid,
+        match=re.escape(
+            "value must be one of ['controlled_load', 'feed_in', 'general'] for dictionary value @ data['channel_type']"
+        ),
+    ):
+        await hass.services.async_call(
+            DOMAIN,
+            GET_FORECASTS_SERVICE,
+            {
+                ATTR_CONFIG_ENTRY_ID: GENERAL_ONLY_SITE_ID,
+                ATTR_CHANNEL_TYPE: "incorrect",
+            },
+            blocking=True,
+            return_response=True,
+        )
+
+
+@pytest.mark.usefixtures("mock_amber_client_general_forecasts")
+async def test_unavailable_channel_type(
+    hass: HomeAssistant,
+    general_channel_config_entry: MockConfigEntry,
+) -> None:
+    """Test error when the channel type is not found."""
+    await setup_integration(hass, general_channel_config_entry)
+
+    with pytest.raises(
+        ServiceValidationError, match="There is no controlled_load channel at this site"
+    ):
+        await hass.services.async_call(
+            DOMAIN,
+            GET_FORECASTS_SERVICE,
+            {
+                ATTR_CONFIG_ENTRY_ID: GENERAL_ONLY_SITE_ID,
+                ATTR_CHANNEL_TYPE: "controlled_load",
+            },
+            blocking=True,
+            return_response=True,
+        )
+
+
+@pytest.mark.usefixtures("mock_amber_client_forecasts")
+async def test_service_entry_availability(
+    hass: HomeAssistant,
+    general_channel_config_entry: MockConfigEntry,
+) -> None:
+    """Test the services without valid entry."""
+    general_channel_config_entry.add_to_hass(hass)
+    mock_config_entry2 = MockConfigEntry(domain=DOMAIN)
+    mock_config_entry2.add_to_hass(hass)
+    await hass.config_entries.async_setup(general_channel_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    with pytest.raises(ServiceValidationError, match="Mock Title is not loaded"):
+        await hass.services.async_call(
+            DOMAIN,
+            GET_FORECASTS_SERVICE,
+            {
+                ATTR_CONFIG_ENTRY_ID: mock_config_entry2.entry_id,
+                ATTR_CHANNEL_TYPE: "general",
+            },
+            blocking=True,
+            return_response=True,
+        )
+
+    with pytest.raises(
+        ServiceValidationError,
+        match='Integration "amberelectric" not found in registry',
```

Comment:
I think this error is strange, because we mean to say that we can't find the config entry id, but now we just say that whole amberelectric is not found, which is incorrect

Reply:
Changed to 'Config entry'

Reply:
But I would rather see that we don't add the domain to the placeholders, but rather something that means something for the user

Reply:
How about just `{target} not found in the registry`?

Reply:
Oh, ok - I see where the confusion is coming from - I've used amberelectric as the config entry in the test.